src/safeposix/syscalls/fs_calls.rs

```diff
@@ -4239,44 +4295,49 @@ impl Cage {
     }
 
     //##------------------SEMAPHORE SYSCALLS------------------
-/*
- *  Initialize semaphore object SEM to value
- *  pshared used to indicate whether the semaphore is shared in threads (when equals to 0)
- *  or shared between processes (when nonzero)
- */
-/// ## `sem_init_syscall`
-///
-/// ### Description
-/// This function initializes a semaphore object, setting its initial value and 
-///specifying whether it's shared between threads or processes.
-/// 1. Boundary Check: The function first checks if the initial value is 
-/// within the allowed range.
-/// 2. Check for Existing Semaphore: The function then checks if a semaphore 
-/// with the given handle already exists.
-/// 3. Initialize New Semaphore: If the semaphore does not exist, the function 
-/// creates a new semaphore object and inserts it into the semaphore table.
-/// 4. Add to Shared Memory Attachments (if shared): If the semaphore is shared 
-/// between processes, 
-/// the function adds it to the shared memory attachments of other processes 
-/// that have already attached to the shared memory segment.
-/// 5. The function ensures thread safety by using a unique semaphore handle and 
-/// checking for existing entries in the semaphore table before attempting to create a new one.
-/// The code also avoids inserting a semaphore into the same cage twice during the shared 
-/// memory attachment process by excluding the initial cage from the iteration loop.
-///[sem_init](https://man7.org/linux/man-pages/man3/sem_init.3.html)
-/// ### Function Arguments
-/// * `sem_handle`: A unique identifier for the semaphore.
-/// * `pshared`:  Indicates whether the semaphore is shared between 
-/// threads (0) or processes (non-zero).
-/// * `value`: The initial value of the semaphore.
-///
-/// ### Returns
-/// * 0 on success.
-/// ### Errors
-/// * `EBADF (9)`: If the semaphore handle is invalid or the semaphore is already initialized.
-/// * `EINVAL (22)`: If the initial value exceeds the maximum allowable value 
-/// * 'ENOSYS(38)' : Currently not supported
-/// for a semaphore (SEM_VALUE_MAX).
+    /*
```

Comment:
- These changes below are just reformatting changes generated by the `cargo fmt` command.
- No new changes have been made to these syscalls.